operators/src/source/ogr_source.rs

```diff
@@ -918,13 +956,120 @@ where
+                        field_value: v,
+                    })?, // TODO: handle other types
+                    Err(e) => error_spec.on_error(Error::Gdal { source: e })?,
+                };
+
+                Ok(FeatureDataValue::NullableFloat(value_option))
+            }
+            FeatureDataType::Int => {
+                #[allow(clippy::match_same_arms)]
+                let value_option = match field {
+                    Ok(Some(FieldValue::IntegerValue(v))) => Some(i64::from(v)),
+                    Ok(Some(FieldValue::Integer64Value(v))) => Some(v),
+                    Ok(Some(FieldValue::StringValue(s))) => i64::from_str(&s).ok(),
+                    Ok(Some(FieldValue::RealValue(v))) => Some(v as i64),
+                    Ok(None) => None,
+                    Ok(Some(v)) => error_spec.on_error(Error::OgrColumnFieldTypeMismatch {
+                        expected: "Int".to_string(),
+                        field_value: v,
+                    })?, // TODO: handle other types
+                    Err(e) => error_spec.on_error(Error::Gdal { source: e })?,
+                };
+
+                Ok(FeatureDataValue::NullableInt(value_option))
+            }
+            FeatureDataType::Category => {
+                #[allow(clippy::match_same_arms)]
+                let _value_option: Option<u8> = match field {
+                    Ok(None) => None,
+                    Ok(Some(v)) => error_spec.on_error(Error::OgrColumnFieldTypeMismatch {
+                        expected: "Category".to_string(),
+                        field_value: v,
+                    })?, // TODO: handle other types
+                    Err(e) => error_spec.on_error(Error::Gdal { source: e })?,
+                };
+
+                Err(Error::InvalidFeatureDataType) // TODO: ?
+            }
+            FeatureDataType::Bool => {
+                #[allow(clippy::match_same_arms)]
+                let value_option = match field {
+                    Ok(Some(FieldValue::IntegerValue(v))) => Some(v != 0),
+                    Ok(Some(FieldValue::Integer64Value(v))) => Some(v != 0),
+                    Ok(Some(FieldValue::StringValue(s))) => bool::from_str(&s).ok(),
+                    Ok(Some(FieldValue::RealValue(v))) => Some(v != 0.0),
+                    Ok(None) => None,
+                    Ok(Some(v)) => error_spec.on_error(Error::OgrColumnFieldTypeMismatch {
+                        expected: "Bool".to_string(),
+                        field_value: v,
+                    })?, // TODO: handle other types
+                    Err(e) => error_spec.on_error(Error::Gdal { source: e })?,
+                };
+
+                Ok(FeatureDataValue::NullableBool(value_option))
+            }
+            FeatureDataType::DateTime => {
+                #[allow(clippy::match_same_arms)]
+                let value_option = match field {
+                    Ok(None) => None,
+                    Ok(Some(v)) => match time_attribute_parser(v) {
```

Comment:
why is there no `OgrColumnFieldTypeMismatch`?

Reply:
The parser generated by `create_time_parser` is a function which takes an owned value. To put the FieldValue inside the `Error::OgrColumnFieldTypeMismatch` i would have to clone the value each time.
The parser currently returns the custom errors `OgrFieldValueIsNotDateTime`, `OgrFieldValueIsNotString` and `OgrFieldValueIsNotValidForSeconds`. Should I change `create_time_parser` to generate `OgrColumnFieldTypeMismatch`, implement Clone on FieldValue or leave it as is?